### PR TITLE
add helper func to read csv file

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: src/spark/utils.py [hunk 1]
```diff
@@ -122,3 +126,52 @@ def get_spark_session(
...
+        path: str,
+        header: bool = True,
+        sep: str = None,
+        **kwargs
+) -> DataFrame:
+    """
+    Read a file in CSV format from minIO into a Spark DataFrame.
+
+    :param spark: The Spark session.
+    :param path: The minIO path to the CSV file.
```
Comment (MrCreosote) [inline src/spark/utils.py:159]: An example would help here given the parsing below

File: src/spark/utils.py [hunk 2]
```diff
@@ -122,3 +126,52 @@ def get_spark_session(
...
+    :return: A DataFrame.
+    """
+
+    if not sep:
+        client = get_minio_client()
+        bucket, key = path.replace("s3a://", "").split("/", 1)
+        obj = client.get_object(bucket, key)
+        sample = obj.read(1024).decode()
+        sep = _detect_delimiter(sample)
+        print(f"Detected delimiter: {sep}")
```
Comment (MrCreosote) [inline src/spark/utils.py:173]: This can return None. What happens in that case?

File: src/spark/utils.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/spark/utils.py]: 👍

File: src/spark/utils.py (hunk 2 above)
Comment (Tianhao-Gu) [reply src/spark/utils.py]: spark will use the default delimiter `,`. Added comments for this behavior as well.

Comment (Tianhao-Gu) [reply src/spark/utils.py]: maybe we should just terminate if we cannot decide and having users to manually put the delimiter themselves?

Comment (MrCreosote) [reply src/spark/utils.py]: I'd guess if the sniffer can't figure it out it's probably not `,`, so terminating seems safer

Comment (MrCreosote) [reply src/spark/utils.py]: Unless 1024 bytes isn't enough to figure it out - first column is really big for example

File: src/spark/utils.py (hunk 1 above)
Comment (MrCreosote) [reply src/spark/utils.py]: :+1:

File: src/spark/utils.py [hunk 3]
```diff
@@ -122,3 +126,52 @@ def get_spark_session(
...
+
+    :return: The detected delimiter or raise error if it cannot be detected
+    """
+
+    try:
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(sample)
+        return dialect.delimiter
+    except Exception as e:
+        raise ValueError(f"Delimiter could not be detected: {e}. Please provide the delimiter explicitly.")
```
Comment (MrCreosote) [inline src/spark/utils.py:145]:
```suggestion
        raise ValueError(
            f"Delimiter could not be detected: {e}. Please provide the delimiter explicitly.") from e
```

File: src/spark/utils.py [hunk 4]
```diff
@@ -122,3 +126,52 @@ def get_spark_session(
...
+    :param kwargs: Additional arguments to pass to spark.read.csv.
+
+    :return: A DataFrame.
+    """
+
+    if not sep:
+        client = get_minio_client()
+        bucket, key = path.replace("s3a://", "").split("/", 1)
+        obj = client.get_object(bucket, key)
+        sample = obj.read(1024).decode()
```
Comment (MrCreosote) [inline src/spark/utils.py:171]:
```suggestion
        sample = obj.read(8192).decode()
```

File: src/spark/utils.py (hunk 2 above)
Comment (MrCreosote) [reply src/spark/utils.py:173]: :+1:

**Review by MrCreosote — APPROVED** [review]
assuming tests pass

**Pull request merged**